library/std/src/fs.rs

```diff
@@ -358,32 +358,26 @@ impl File {
         OpenOptions::new().write(true).create(true).truncate(true).open(path.as_ref())
     }
 
-    /// Returns a new OpenOptions object.
+    /// Returns a builder struct to configure how a file is opened.
     ///
-    /// This function returns a new OpenOptions object that you can use to
-    /// open or create a file with specific options if `open()` or `create()`
-    /// are not appropriate.
+    /// This function returns an empty [`OpenOptions`] object
+    /// to avoid the need of importing [`OpenOptions`].
     ///
-    /// It is equivalent to `OpenOptions::new()` but allows you to write more
-    /// readable code. Instead of `OpenOptions::new().read(true).open("foo.txt")`
-    /// you can write `File::with_options().read(true).open("foo.txt")`. This
-    /// also avoids the need to import `OpenOptions`.
-    ///
-    /// See the [`OpenOptions::new`] function for more details.
+    /// See [`OpenOptions::new`] function for more details.
     ///
     /// # Examples
     ///
     /// ```no_run
-    /// #![feature(with_options)]
+    /// #![feature(file_builder)]
     /// use std::fs::File;
     ///
     /// fn main() -> std::io::Result<()> {
-    ///     let mut f = File::with_options().read(true).open("foo.txt")?;
+    ///     let _ = File::builder().read(true).open("foo.txt")?;
     ///     Ok(())
     /// }
     /// ```
-    #[unstable(feature = "with_options", issue = "65439")]
-    pub fn with_options() -> OpenOptions {
+    #[unstable(feature = "file_builder", issue = "65439")]
+    pub fn builder() -> OpenOptions {
```

Comment:
I wonder why not `options`?

Reply:
I guess I've never seen APIs like that. But yeah, it could be if reviewer prefer.

Reply:
Based on the informal poll in #65439, `builder` was the least popular name and `options` was the most popular (just).

Reply:
I know that. However Rust design has never been a community vote, but
it depends much on official respective teams.

Reply:
Right, just pointing it out in case whoever reviews this wasn't aware.

Reply:
Wow, I didn't even know there is a vote. I just came out with it from my mind. Good to know that we are on the same page and same taste. :D